src/flatbuffer_types.rs

```diff
@@ -31,6 +33,17 @@ pub(crate) struct SsTableInfoOwned {
     data: Bytes,
 }
 
+impl Serialize for SsTableInfoOwned {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut s = serializer.serialize_struct("SsTableInfoOwned", 2)?;
+        s.serialize_field("data_length", &self.data.len())?;
```

Comment:
is it helpful to hexify the data itself and include it here? - for now I just included the data length but I'm not sure that's at all helpful

cc @rodesai @criccomini

Reply:
It would be useful to include the sst metadata. I don't think there's a great option for doing this right now:
- Ideally we wouldn't be using a flatbuffers type in most places (I filed https://github.com/slatedb/slatedb/issues/213). Once we fix that, the json serde should just serialize this nicely for us.
- Flatbuffers generally supports serializing/deserializing to json. Unfortunately, this is only supported by the C++ API, so we'd need to define FFI to access it, which seems like a pain.

What if for this patch we define our own type for SST metadata and use it here to get a nice json string. And then we can adapt that in the rest of the code base in a follow on PR? I'm happy to leave your patch as is for now too, and do that in a follow-up.

Reply:
👍 let's keep it as is for now and I'll just take on #213 next. I feel that'll be another good intro to the codebase for me, and then I can wrap back around to finish this part up.